todo_app/helpers/decorators.py

```diff
@@ -0,0 +1,20 @@
+from flask import redirect
+from flask_dance.contrib.github import github
+
+
+def login_required(original_route):
+    def protected_route(*args, **kwargs):
+        if not github.authorized:
+            return redirect("/login")
+        return original_route(*args, **kwargs)
+    protected_route.__name__ = original_route.__name__
```

Comment:
A while since I've seen this one - [the functools `wraps` decorator](https://www.geeksforgeeks.org/python-functools-wraps-function/) is a nicer way of capturing this in my opinion:
```
from functools import wraps

def login_required(original_route):
    @wraps(original_route)
    def protected_route(*args, **kwargs):
        if not github.authorized:
            return redirect("/login")
        return original_route(*args, **kwargs)
    return protected_route
```